public/javascripts/components/appCardsAndCategory.js

```diff
@@ -0,0 +1,95 @@
+app.component('homepage-base', {
+  data() {
+    return {
+      apps: [],
+      appsIsEmpty: false,
+      categories: []
+    }
+  },
+
+  created: function () {
+    fetch(`/api/categories`)
+      .then(response => response.json())
+      .then(res => {
+        this.categories = res.data.categories;
+      }).then(() => {
+        fetch(`/api/category/all`)
+          .then(response => response.json())
+          .then(json => {
+            this.apps = json.data.apps;
+          });
+      });
+  },
+
+  methods: {
+    getActive(categoryName) {
+      let selectedElement;
+      this.categories.forEach((category) => {
+        let element = document.getElementById(category.name);
+        //check which category is active and which are not, adjust class name
+        if (categoryName !== category.name) {
+          element.classList.add('inactive');
+        }
+        if (categoryName === category.name) {
+          selectedElement = element; // save selected category for toggle filter
+          element.classList.remove('inactive');
+        }
+      });
+      // if button doesnt have 'selected' class, add selected, run category filter
+      if (!selectedElement.classList.contains("selected")) {
+        selectedElement.classList.add("selected");
+        fetch("/api/category/" + categoryName + '?format=json')
```

Comment:
oh i see, does it fetch then the separate category instead of all of them?

Reply:
this part is to filter apps by category 👍